### PR TITLE
VSSPP-445 (Support 'custom' adRenderer value)

### PR DESCRIPTION
**Pull request by ynisman**

### CODE REVIEW

**Comment by ynisman** [comment]
> We need to make changes to the README page to include the instructions about specifying the 'custom' adRenderer value. We also need to include in the file where to change the code so that if they are using 'custom', that they do not get the "not implemented in this version" message

Modofied readme file.

File: README.md [hunk 1]
```diff
@@ -129,14 +129,18 @@ If you have either created your own build of the MailOnline plugin or are replac
 
 #### Using your own rendering code
 
-If you are replacing the MailOnline rendering plugin with your own custom rendering code, you need to change the code where currently the MailOnline plugin is loaded and invoked.
+If you are replacing the MailOnline rendering plugin with your own custom rendering code, you need to change the code where currently an ad renderer is loaded and invoked.
 
+- You have to explicitly set `adRendere` options to `custom` value.
```
Comment (graphicdeveloper) [inline README.md]: typo: `adRendere` >> should be: `adRenderer`.
Also, I would reword it to read like this:
You have to explicitly add the  `adRenderer` param to your configuration object with this value: `adRenderer: "custom"`.

File: README.md [hunk 2]
```diff
@@ -129,14 +129,18 @@ If you have either created your own build of the MailOnline plugin or are replac
 
 #### Using your own rendering code
 
-If you are replacing the MailOnline rendering plugin with your own custom rendering code, you need to change the code where currently the MailOnline plugin is loaded and invoked.
+If you are replacing the MailOnline rendering plugin with your own custom rendering code, you need to change the code where currently an ad renderer is loaded and invoked.
 
+- You have to explicitly set `adRendere` options to `custom` value.
 - Your renderer must be compatible with the Brightcove Player environment.
-- Remove the following function, which is used to load the MailOnline plugin: `./src/BcPrebidVast.js>loadMolPlugin(callback)`
+- Add an function to load your renderer in `./src/BcPrebidVast.js`.
```
Comment (graphicdeveloper) [inline README.md]: Typo: Should be "a" - not "an" - "Add **a** function to your renderer..."

File: README.md [hunk 3]
```diff
@@ -129,14 +129,18 @@ If you have either created your own build of the MailOnline plugin or are replac
...
 #### Using your own rendering code
 
-If you are replacing the MailOnline rendering plugin with your own custom rendering code, you need to change the code where currently the MailOnline plugin is loaded and invoked.
+If you are replacing the MailOnline rendering plugin with your own custom rendering code, you need to change the code where currently an ad renderer is loaded and invoked.
 
+- You have to explicitly set `adRendere` options to `custom` value.
 - Your renderer must be compatible with the Brightcove Player environment.
-- Remove the following function, which is used to load the MailOnline plugin: `./src/BcPrebidVast.js>loadMolPlugin(callback)`
+- Add an function to load your renderer in `./src/BcPrebidVast.js`.
+- If you don't want to use MailOnline plugin, remove the following function, which is used to load the MailOnline plugin: `./src/BcPrebidVast.js>loadMolPlugin(callback)`
```
Comment (graphicdeveloper) [inline README.md]: Missing "the" here -- should be: "If you don't want to use **the** MailOnline plugin..."

File: README.md [hunk 4]
```diff
@@ -129,14 +129,18 @@ If you have either created your own build of the MailOnline plugin or are replac
...
+  - Make sure that all calls to this function have also been removed from your plugin code.
+- If you don't want to use IMA plugin, remove the following function, which is used to load the IMA plugin: `./src/BcPrebidVast.js>loadImaPlugin(callback)`
```
Comment (graphicdeveloper) [inline README.md]: Missing "the" here -- should be: "If you don't want to use **the** IMA plugin..."

File: src/BcPrebidVast.js [hunk 5]
```diff
@@ -914,11 +917,19 @@ var prebidVastPlugin = function (player) {
 				if (_adRenderer === 'ima') {
```
Comment (graphicdeveloper) [inline src/BcPrebidVast.js]: Shouldn't you be comparing to the constants in this if/else stack, instead of the strings themselves (`_rendererNames.IMA`, not `'ima'` -- `_rendererNames.CUSTOM`, not `'custom'`)

File: src/VastManager.js [hunk 6]
```diff
@@ -347,6 +347,15 @@ var vastManager = function () {
 					}
 					_vastRendererObj.playAd(xml, _options, firstVideoPreroll, _mobilePrerollNeedClick, prerollNeedClickToPlay, eventCallback);
 				}
+				else if (_options.adRenderer === 'custom') {
```
Comment (graphicdeveloper) [inline src/VastManager.js]: Again, shouldn't you be comparing to a constant here, not the string (`_rendererNames.CUSTOM`, not `'custom'`)?

**Comment by ynisman** [comment]
> typo - should be adRenderer. Also the sentence should be:
done

File: src/BcPrebidVast.js (hunk 5 above)
Comment (ynisman) [reply src/BcPrebidVast.js]: Created Contants.js file and use renderer names from this file.
